api/src/test/java/org/openmrs/module/eptsreports/reporting/unit/utils/EptsCalculationUtilsTest.java

```diff
@@ -0,0 +1,378 @@
+import org.openmrs.calculation.patient.PatientCalculationService;
+import org.openmrs.calculation.patient.PatientCalculationServiceImpl;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.ListResult;
+import org.openmrs.calculation.result.ObsResult;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
+import org.openmrs.module.eptsreports.reporting.cohort.definition.JembiObsDefinition;
+import org.openmrs.module.eptsreports.reporting.helper.TestsHelper;
+import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterDatetimeDataDefinition;
+import org.openmrs.module.reporting.data.patient.EvaluatedPatientData;
+import org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition;
+import org.openmrs.module.reporting.data.patient.service.PatientDataService;
+import org.openmrs.module.reporting.data.person.EvaluatedPersonData;
+import org.openmrs.module.reporting.data.person.definition.ObsForPersonDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PersonDataDefinition;
+import org.openmrs.module.reporting.data.person.service.PersonDataService;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.test.BaseContextMockTest;
+
+public class EptsCalculationUtilsTest extends BaseContextMockTest {
+
+  @Mock private PatientCalculationService service;
+
+  @Spy private TestsHelper testsHelper;
+
+  @Mock private PatientDataService patientDataService;
+
+  @Mock private PersonDataService personDataService;
+
+  @Mock private ServiceContext serviceContext;
+
+  private PatientCalculationContext calculationContext;
+
+  @SuppressWarnings("deprecation")
+  @Before
+  public void init() {
+    contextMockHelper.setService(PersonDataService.class, personDataService);
+    contextMockHelper.setService(PatientDataService.class, patientDataService);
+    when(service.createCalculationContext())
+        .thenReturn(new PatientCalculationServiceImpl().new SimplePatientCalculationContext());
+    calculationContext = service.createCalculationContext();
+    calculationContext.setNow(testsHelper.getDate("2019-05-30 00:00:00.0"));
+  }
+
+  /**
+   * unit tests {@link EptsCalculationUtils#ensureEmptyListResults(CalculationResultMap,
+   * java.util.Collection)}
+   */
+  @Test
+  public void ensureEmptyListResultsShouldReplaceNullsWithEmptyList() {
+    CalculationResultMap map = new CalculationResultMap();
+    ListResult list = new ListResult();
+    map.put(1, list);
+    map.put(2, null);
+    map.put(3, list);
+    Assert.assertNull(map.get(2));
+    CalculationResultMap replacedMap =
+        EptsCalculationUtils.ensureEmptyListResults(map, Arrays.asList(1, 2, 3));
```

Comment:
Remember to add some test for those edge cases (pararmeters are null etc...)

Reply:
of-couse this won't test NPE, closely examine each entry in the map since each represents a scenario